sematic/scheduling/kubernetes.py

```diff
@@ -26,19 +28,32 @@
     ResourceRequirements,
 )
 from sematic.scheduling.external_job import KUBERNETES_JOB_KIND, ExternalJob, JobType
+from sematic.utils.exceptions import ExceptionMetadata
 from sematic.utils.retry import retry
 
 logger = logging.getLogger(__name__)
 _kubeconfig_loaded = False
 
-
+POD_PHASE_PRECEDENCE = ["Pending", "Running", "Succeeded", "Failed", "Unknown"]
```

Comment:
note: my comment is assuming your precedence is "if we somehow saw multiple of these, which would we want to surface as 'the truth'." It appears it might mean "what order are they likely to happen in," in which case I agree with everything, except that "Unknown" is a bit unclear where to go.

Reply:
Added comments describing the logic and linked a resource.
You were right in your understanding. Initially I left "Uknown" last in case there were some zombie pods, and K8 created a new pod to replace them, but I realized this was just wishful thinking. "Unknown" is not clearly documented, so we must assume the worst. Moved it as the first in the list.